File: ios-native/MomentsAPIWebDemoApp/MomentsAPIWebDemoApp/Config/AppConfig.swift
//
//  AppConfig.swift
//  MomentsAPIWebDemoApp
//
//  Created by shivang vyas on 07/05/25.
//

import Foundation

/// Configuration constants for the app
enum AppConfig {
    /// API related configuration
    enum API {
        /// Base URL for the offers API endpoint
        static let baseURL = "https://api.adspostx.com/native/v4/offers.json"
        
        /// Default SDK ID for demo purposes
        static let defaultSDKId = "<use_your_sdkid_here>"
    }
    
    /// WebView related configuration
    enum WebView {
        /// Base URL for loading HTML content
        static let baseURL = Bundle.main.bundleURL
        
        /// CDN URL for the WebSDK launcher script
        static let launcherScriptURL = "https://cdn.adspostx.com/launcher.min.js"
    }
}

File: flutter/MSSDKDemoApp-Flutter/mssdk_demo_app/lib/config/app_config.dart
import 'package:flutter_dotenv/flutter_dotenv.dart';

/// Configuration constants for the app
class AppConfig {
  /// API related configuration
  static final api = _API();

  /// WebView related configuration
  static final web = _Web();

  /// Load environment variables
  static Future<void> load() async {
    await dotenv.load();
  }
}

/// API configuration constants
class _API {
  /// Base URL for the offers API endpoint
  String get baseURL => dotenv.env['API_BASE_URL'] ?? "";

  /// Default SDK ID for demo purposes
  String get defaultSDKId => dotenv.env['API_DEFAULT_SDK_ID'] ?? "";
}

/// WebView configuration constants
class _Web {
  /// CDN URL for the WebSDK launcher script
  String get launcherScriptURL => dotenv.env['WEBSDK_LAUNCHER_URL'] ?? "";
}
